network/p2p/builder.go

```diff
@@ -138,10 +139,31 @@ type NodeBuilder interface {
 	Build() (LibP2PNode, error)
 }
 
-// PeerScoringConfig is a configuration for peer scoring parameters for a GossipSub pubsub system.
-type PeerScoringConfig struct {
+// PeerScoringConfigOverride is a structure that is used to carry over the override values for peer scoring configuration.
+// Any attribute that is set in the override will override the default peer scoring config.
+// Typically, we are not recommending to override the default peer scoring config in production unless you know what you are doing.
+type PeerScoringConfigOverride struct {
 	// TopicScoreParams is a map of topic score parameters for each topic.
+	// Override criteria: any topic (i.e., key in the map) will override the default topic score parameters for that topic and
+	// the corresponding value in the map will be used instead of the default value.
+	// If you don't want to override topic score params for a given topic, simply don't include that topic in the map.
+	// If the map is nil, the default topic score parameters are used for all topics.
 	TopicScoreParams map[channels.Topic]*pubsub.TopicScoreParams
+
 	// AppSpecificScoreParams is a function that returns the application specific score parameters for a given peer.
+	// Override criteria: if the function is not nil, it will override the default application specific score parameters.
+	// If the function is nil, the default application specific score parameters are used.
 	AppSpecificScoreParams func(peer.ID) float64
+
+	// DecayInterval is the interval over which we decay the effect of past behavior. So that
+	// a good or bad behavior will not have a permanent effect on the penalty. It is also interval
+	// that GossipSub refreshes the scores of all peers.
```

Comment:
Feel free to edit with different ways.
```suggestion
	// DecayInterval is the interval over which we decay the effect of past behavior, so that
	// a good or bad behavior will not have a permanent effect on the penalty. It is also the interval
	// that GossipSub uses to refresh the scores of all peers.
```